gui/wxpython/core/gconsole.py

```diff
@@ -529,14 +544,16 @@ def RunCmd(
             ):
                 event = gIgnoredCmdRun(cmd=command)
                 wx.PostEvent(self, event)
-                return
 
+                self.UpdateHistory(status=history.STATUS_FAILED)
```

Comment:
I don't think this is right. The status should be updated probably in `RunSpecialCmd` and `RunDisplayCmd`, no? E.g., typing `d.rast` in console should be success, typing `d.raast` should be failure.

Reply:
I agree with the necessary changes in RunSpecialCmd and RunDisplayCmd but it is more complicated. We have basically three variants of command ends.

1) RunSpecialCmd -  commands like _d.rast_, _v.import_ and so on. In the case of d. modules, when I write e.g. _d.raast_, the module fails and you are right that the history update should be implemented in RunSpecialCmd (I implemented it in the last commit hash (yet no refactoring))

2) Commands like _v.what, r.lake_ and similar Non-special commands - when I launch those types of commands without arguments, the module dialog is launched - and this is a problem, the code of OnCmdDone is never reached. This is reason why I think we need to edit gconsole.py. Otherwise, there will still be the "running" status.

3) Commands like _v.whaaat_, _aaaa,_ and other nonsense stuff - not sure how to solve those cases because the status in OnCmdDone always ends up in "success"... so in this case the result is not "running" but always ends up in "success" (still better than the point 2 but weird)

So, not sure how approach points 2 and 3 ..

Reply:
Could we just not add all these to the command history in the first place? So that we don't have to solve this issue.

Reply:
I think not including those commands will be more complicated than just having them here with the wrong status. Moreover, I would say that we need to include all commands that were launched otherwise history is losing its purpose... So I would suggest the following procedure:

Points 1 and 2 are okay after changes in the main frame and in gconsole.

I think that the Point 3 is actually the bug which does not directly relate to this PR but this PR revealed the problem. When any nonsense command is launched, it should end up in `event.returncode != 0 `on the row 767 in gconsole.py. Currently it ends up in "Success" which is not correct.
I am not sure how to fix it though. It would require many changes in how command status is managed...
 
I think that still it is not too big problem - I will create the Issue for it and I think despite this issue this PR can go to the main.

What do you think @petrasovaa? Is that acceptable?

Reply:
> I think excluding those commands will be more complicated than achieving the correct status. Moreover, I would say that we need to include all commands that were launched otherwise history is losing its purpose... So I would suggest the following procedure:
> 
> Points 1 and 2 are okay after changes in the main frame and in gconsole.
> 
> I think that the Point 3 is actually the bug which does not directly relate to this PR but this PR revealed the problem. When any nonsense command is launched, it should end up in `event.returncode != 0 `on the row 767 in gconsole.py. Currently it ends up in "Success" which is not correct. I am not sure how to fix it though. It would require many changes in how command status is managed...
> 
> I think that still it is not too big problem - I will create the Issue for it and I think despite this issue this PR can go to the main.
> 
> What do you think @petrasovaa? Is that acceptable?

Reply:
Definitely, that's fine, this is almost ready.

Reply:
Noted here https://github.com/OSGeo/grass/issues/3698 .